### ISSUE TITLE
Poetry install gets killed OOM 

### ISSUE THREAD

**Issue by rednaks** [python-poetry/poetry#3756]
- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: ubuntu 20.04.1
- **memory**:  512M, free: 253M
- **Poetry version**: 1.1.5
logs: 
```
Installing dependencies from lock file

Package operations: 60 installs, 0 updates, 0 removals

  • Installing asgiref (3.3.1): Installing...
  • Installing numpy (1.20.1): Installing...
  • Installing pycparser (2.20): Installing...
  • Installing pytz (2021.1): Installing...
  • Installing sqlparse (0.4.1): Installing...
  • Installing vine (5.0.0): Installing...
Killed

```

## Issue
Noticed that today my server deployments fails during the poetry install task. after investigating, figured out that poetry is getting killed by the OS, Out Of Memory. 

I have a machine with very limited resources, it was working fine until I updated poetry. 

After doing some research I figured out that `installer.parallel` was added `true` by default in version `1.1.4`. when I disabled the option the deployments are not failing anymore.

**Comment by abn** [comment]
Out of curiosity, is this machine a development environment or a target environment? 

I suspect that the issue here is that the memory consumption is real since it is in parallel downloading, unpacking and installing (which might involve build/compile for packages like numpy). OS might be killing off the parent process. If this is not a leak, then the workaround that you have listed might be the actual solution for this scenario.

Maybe generating some profiles with something like https://pythonspeed.com/products/filmemoryprofiler/ might help triage the issue.

**Comment by rednaks** [comment]
well, it's a dev machine, I don't have enough ram for everything so I minimized the VMs locally, I'll try the profiler but I'm wondering, how many tasks poetry executes in parallel ? I see here he's trying to install 6 packages simultaneously I would be really concerned if the same thing happens on production and creates 100 process for installing all the dependencies at once.

My machine have 2 cores, poetry could use the number of CPUs (or 2 * CPUs) to execute parallel tasks and of course an option would be welcome.

**Comment by abn** [comment]
I believe you are looking for https://github.com/python-poetry/poetry/blob/129ed0e1c4580683c5699ab780fed233c308fd18/poetry/installation/executor.py#L66-L76.

I suspect we will need a new configuration and update the defaults maybe.

**Comment by rednaks** [comment]
Hello, any news on this issue ? the only solution I'm using right now is to disable `parallel`.
I think it's really dangerous especially when you download modules that compile things in C. if for example `make` is also using parallel jobs `-jX` it becomes a systematic issue ..

**Comment by dimbleby** [comment]
I expect it's compiling numpy that is burning through RAM anyway - there's nothing much poetry can do about that.

If disabling parallel works for you, great, do that.  Or allocate more RAM to your VMs, 512M really isn't very much at all these days.

Pretty sure that there'll be no poetry fix here.

**Comment by neersighted** [comment]
Correct, this is wontfix. We can tune the default parallelism, but at the end of the day if the default is too aggressive for your system resources, that's why we have the knob.
